### ISSUE TITLE
Add essential ValueSets

### ISSUE THREAD

**Issue by costateixeira** [hl7-eu/unicom-ig#32]
We need some valuesets to support the profiles

Minimal CSV should look like :
| code | system | designation |
|---|---|----|
|---|---|----|

More advanced code systems can include things like languages:
|code | system | designation | language 1 | designation in language 1 | language N | designation in language N |
|---|---|----|----|----|----|----|
|---|---|----|----|----|----|----|

and/or hierarchies, polyhierarchies:
|code | system | designation | property type (e.g. "is parent of") | property value (e.g. the code of the parent)|
|---|---|----|----|----|
|---|---|----|----|----|

(properties can be relations to other codes, or text, or dates,...

**Comment by rlindstrm** [comment]
Just looking at this makes me want to go to sleep.
I'll have to wait until my motivation gets shipped.

**Comment by costateixeira** [comment]
if we need valuesets, this is how we make them. This, or manually editing XMLs according to the spec

**Comment by rlindstrm** [comment]
> if we need valuesets, this is how we make them. This, or manually editing XMLs according to the spec

I'm still not an idiot, José. Just lazy.

**Comment by costateixeira** [comment]
Which is why I recommend the CSVs, I am more experienced at being lazy

**Comment by rlindstrm** [comment]
Two/three value sets I need to unblock my profiling:
**1) Authorised Dose Form**
Combines four SPOR lists (they are also separate like this in EDQM). Note, that all SPOR lists are basically separate code systems in FHIR.

- Pharmaceutical Dose Form   | https://spor.ema.europa.eu/rmswi/#/lists/200000000004/terms
- Combined Pharmaceutical Dose Form   | https://spor.ema.europa.eu/rmswi/#/lists/200000000006/terms
- Combined Term   | https://spor.ema.europa.eu/rmswi/#/lists/200000000007/terms
- Combination Package   | https://spor.ema.europa.eu/rmswi/#/lists/200000000008/terms

**2) Unit of Measurement and Unit of Presentation.**
If I can bind two value sets to one element, then I'd keep them separately. 
It's only the presentation strength denominator, where the unit can be either one or the other (60mg/0.6ml or 60mg/vial would both be considered presentation strength if the vial is 0.6ml)

- Unit of Presentation   |  https://spor.ema.europa.eu/rmswi/#/lists/200000000014/terms
- Units of Measurement | https://spor.ema.europa.eu/rmswi/#/lists/100000110633/terms

I believe others need these value sets, too. They're pretty essential.
There are many other SPOR lists, but those I can define as system.value and save myself from handling too many value sets.

And now you tell me to send you the codes... I'll get there. One step at the time. I need a nap now.

**Comment by Censacrof** [comment]
> Two/three value sets I need to unblock my profiling: **1) Authorised Dose Form** Combines four SPOR lists (they are also separate like this in EDQM). Note, that all SPOR lists are basically separate code systems in FHIR.
> 
> * Pharmaceutical Dose Form   | https://spor.ema.europa.eu/rmswi/#/lists/200000000004/terms
> * Combined Pharmaceutical Dose Form   | https://spor.ema.europa.eu/rmswi/#/lists/200000000006/terms
> * Combined Term   | https://spor.ema.europa.eu/rmswi/#/lists/200000000007/terms
> * Combination Package   | https://spor.ema.europa.eu/rmswi/#/lists/200000000008/terms
> 
> **2) Unit of Measurement and Unit of Presentation.** If I can bind two value sets to one element, then I'd keep them separately. It's only the presentation strength denominator, where the unit can be either one or the other (60mg/0.6ml or 60mg/vial would both be considered presentation strength if the vial is 0.6ml)
> 
> * Unit of Presentation   |  https://spor.ema.europa.eu/rmswi/#/lists/200000000014/terms
> * Units of Measurement | https://spor.ema.europa.eu/rmswi/#/lists/100000110633/terms
> 
> I believe others need these value sets, too. They're pretty essential. There are many other SPOR lists, but those I can define as system.value and save myself from handling too many value sets.
> 
> And now you tell me to send you the codes... I'll get there. One step at the time. I need a nap now.

Shouldn't be a problem, you can find the valuesets at this links (normally works, for some reasons RMS is down right now):

1. [Pharmaceutical Dose Form](https://spor.ema.europa.eu/v1/lists/200000000004)
2. [Combined Pharmaceutical Dose Form](https://spor.ema.europa.eu/v1/lists/200000000006)
3. [Combined Term](https://spor.ema.europa.eu/v1/lists/200000000007)
4. [Combination Package](https://spor.ema.europa.eu/v1/lists/200000000008)
5. [Unit Of Presentation](https://spor.ema.europa.eu/v1/lists/200000000014)
6. [Units of Measurement](https://spor.ema.europa.eu/v1/lists/100000110633)

You need to register and login in order to visualize this valuesets.

**Comment by costateixeira** [comment]
Can anyone make CSVs?

**Comment by rlindstrm** [comment]
> Can anyone make CSVs?

EMA SPOR is down. Time to get a life.

**Comment by costateixeira** [comment]
Robert suggests 2 more value sets:
1. Substances with role of PAI - we only have the codes for 35 substances
2. Moiety - we only have the codes for 35 substances
3. ATC

For dose form, 
Giorgio suggests that we look at eHDSI to see if they are aligned / overlapping. Question: EDQM or SPOR? 
We agree to start with EDQM. The ValueSet / CodeSystem is available.

Assignees:
* Pharmaceutical Dose Form (FG)
* Combined Pharmaceutical Dose Form (FG)
* Combined Term (FG)
* Combination Package (FG)
* Unit Of Presentation (FG)
* Units of Measurement (FG)
* ATC (GC)
* Substances (PAI And moiety) (RVS + FG)

**Comment by rlindstrm** [comment]
> * Substances with role of PAI - we only have the codes for 35 substances
> * Moiety - we only have the codes for 35 substances
> * ATC

What is PAI? Primary active ingredient? 
As far as I know, you cannot differentiate active ingredients, moieties or excipience in SMS value set. All you get is a value set of 50 000 concepts (available in eHDSI, not the newest version but should work ok). Role of the substance is defined in medication data.

eHDSI ATC and EDQM value sets should be ok, we included combination package dose forms quite recently. But they don't have SPOR codes.
Unit of Measurement in eHDSI and SPOR have slight differences, which I'd like to have a rant about.

**Comment by costateixeira** [comment]
Thank Giorgio

This being in the hands of HL7 and since it is in the terminology server,
perhaps we can do it intensionally.

Agree?

**Comment by rlindstrm** [comment]
> Agree?

It seems to be a version that is 2 years old, but the big changes took place in 2020 and seem to be in. Ok for piloting, but they should really update it every year.

And EMA uses SPOR codes instead of ATC codes. Just saying, and hoping you would understand and document how this will be a problem one day:
![image](https://user-images.githubusercontent.com/112903984/199987446-fee7f205-aad1-49ba-a1cf-f876218d5a1d.png)

**Comment by gcangioli** [comment]
This was my idea as well. Define the value set intentionally makes it independent from the version (it is a matter of expansion)

We can reuse this value set https://build.fhir.org/ig/HL7/fhir-ips/ValueSet-whoatc-uv-ips.html or just copy the content and have it in our guide as a new value set

**Comment by Censacrof** [comment]
> What is PAI? Primary active ingredient?
> As far as I know, you cannot differentiate active ingredients, moieties or excipience in SMS value set. All you get is a value set of 50 000 concepts (available in eHDSI, not the newest version but should work ok). Role of the substance is defined in medication data.

PAI is for Precise Active Ingredient. If I understood correctly, a substance has the role of Precise Active Ingredient when it's used as the only ingredient of a ManufacturedItem. I agree with you that it makes no sense to have a valueset for that since IDMP has the "role" attribute inside **Ingredient.**

**Comment by gcangioli** [comment]
> <..> i would copy the content of the IPS to our IG so that we can mess with it as we want. If you agree, do you want to do it? 

added https://build.fhir.org/ig/hl7-eu/unicom-ig/ValueSet-whoatc-unicom.html

**Comment by rlindstrm** [comment]
I just realised, that some of the RMS value sets I need are defined in 5.0.0 Ballot resources as Example value sets (they were not there for Snapshot, not sure when they appeared).
Anyway.. Some are clearly incomplete and useless, some might be usable. How would I go about them? Can I use them as value sets or should I just steal the content and define my own?
For example, they have in specification:  
http://hl7.org/fhir/2022Sep/codesystem-legal-status-of-supply.html
And EMA has defined it at some point as:
https://spor.ema.europa.eu/rmswi/#/lists/100000072051/terms | Legal Status of Supply

EMA does not have their lists published in FHIR format, and I'm a little confused which value set and code system I would use for profiles - the Example one that exists, or the official one that doesn't?

**Comment by rlindstrm** [comment]
Maturity level 1, Trial use, Any context, Case sensitive (!). 
And no indication that the codes are from EMA SPOR RMS. 
https://build.fhir.org/codesystem-unit-of-presentation.html
https://build.fhir.org/codesystem-administrable-dose-form.html
Please help me understand what to make of this.

**Comment by costateixeira** [comment]
The HL7 community could not publish EMA valusets because they were not
openly available (strategy miss from EMA, IMO. They could have the FHIR
world looking at their values if they had presented those ValueSets).
And the FHIR specs are always expected to have some value set for codeable
elements. This was not there in R5.0.0-snapshot*1 *(we should get used to
say snapshot1, because in a few weeks there'll be a snapshot2). So
R5.0.0-ballot fixed that by adding any value sets just to give some
examples.
The binding strength is example so everyone can use different valuesets.

In practice: Just create new ones, stealing as you need.

**Comment by costateixeira** [comment]
Thanks @Censacrof 
I had a different approach (from CSVs) but what matters is to have the valuesets.
We can perhaps put this in a branch to get it validated - we need to see where. For this, 2 remarks:
1. I guess these are the "original" EMA ones - we had decided for using the "global" codes. We can use the different slices for the regulatory part, and perhaps for the MAL?
@rlindstrm I think you are the best to validate if these code systems pass the smell test, once we publish them in the IG - and hopefully you can advise where we use them (I presume in the corresponding slices in the profiles)

**Comment by rlindstrm** [comment]
@costateixeira These are EDQM ones, not EMA. 
Looks pretty advanced, @Censacrof  :)

**Comment by Censacrof** [comment]
> @costateixeira These are EDQM ones, not EMA. Looks pretty advanced, @Censacrof :)

@rlindstrm yep, these are only the EDQM ones, all of them, even the ones that we probably won't need. This was my first time working with SUSHI, please let me know if I need to change or improve anything.

**Comment by Censacrof** [comment]
> Thanks @Censacrof I had a different approach (from CSVs) but what matters is to have the valuesets. We can perhaps put this in a branch to get it validated - we need to see where.

I branched from master but I think that GitHub allows you to merge my pull request in a new branch if you want.

**Comment by rlindstrm** [comment]
[UCUM in eHDSI](https://art-decor.ehdsi.eu/art-decor/decor-valuesets--epsos-?id=1.3.6.1.4.1.12559.11.10.1.3.1.42.16&effectiveDate=2021-04-19T16:20:00&language=en-US)

**Comment by costateixeira** [comment]
11/11 Francesco has done all the valuesets except UCUM - will get the info from Rutt's comment above.
Robert will continue with Substances
José will merge Giorgio's ATC values

**Comment by Censacrof** [comment]
I also added the CodingSystem for the UcumUnits to the pull request #36 . Please, review it so that we can merge it.


> [UCUM in eHDSI](https://art-decor.ehdsi.eu/art-decor/decor-valuesets--epsos-?id=1.3.6.1.4.1.12559.11.10.1.3.1.42.16&effectiveDate=2021-04-19T16:20:00&language=en-US)
> 
> I also have the whole value set catalogue in Excel, if you need. Contains EMA SMS codes as well.

Thanks! I think that will be useful to create a mapping between UCUM and Spor

**Comment by rlindstrm** [comment]
[MVC_610-ainult-ravimid-20220706.xlsx](https://github.com/hl7-eu/unicom-ig/files/10005172/MVC_610-ainult-ravimid-20220706.xlsx)
This is eHDSI master value set catalogue for wave 6. This file only contains prescription-specific value sets.
UCUM-SPOR mappings are theoretically present in RMS Unit of Measurement list, not too easy to extract, but I think you'll manage (I know I wouldn't).

**Comment by rlindstrm** [comment]
Robert's value set for PPL substances with mappings and divisions into categories (see different sheets) [SubtancesCodeTemplate_RVS_V5.xlsx](https://github.com/hl7-eu/unicom-ig/files/10020372/SubtancesCodeTemplate_RVS_V5.xlsx)

Some SMS codes are missing from the file, but I can see they exist in the eHDSISubstance value set based on SMS, so I think this won't be a problem.
The email discussion was about if this value set can be created and maintained for production use. At this point it cannot, so we have to keep an eye on SRS, SMS and SNOMED developments.

**Comment by costateixeira** [comment]
Which code do we stick to in the ValueSet? I think we need to pick one for now - right @gcangioli ?

**Comment by rlindstrm** [comment]
[eP guidelines ](https://health.ec.europa.eu/system/files/2022-06/ehn_guidelines_eprescriptions_en.pdf)say SMS. But you can have several value sets and make use of the mappings Robert has created.

**Comment by costateixeira** [comment]
Discussion: For the codesystems that are not available (i.e. they are behind EMA login), we can create a CodeSystem (Supplement) and use that. At the end we need ValueSets referring to valid CodeSystems, even if the CodeSystems are themselves "temporary local copies".
**Action: Giorgio to validate the code systems and see whether they need to be simply changed to ValueSets.**
@gcangioli this: https://github.com/hl7-eu/unicom-ig/pull/36/files

**Comment by gcangioli** [comment]
@Censacrof : the code systems I looked at are value sets.
In order to keep  the different languages, unfortunately, you cannot simply change the declaration from CodeSystem to ValueSet and removing the definitions.
 
attached an example of how it can be done in sushi (just for one coded concept)
[fsh-example.txt](https://github.com/hl7-eu/unicom-ig/files/10135279/fsh-example.txt)

Hope this helps :-)

**Comment by gcangioli** [comment]
this is how is shown in the guide
![image](https://user-images.githubusercontent.com/35879607/205135369-95b97132-bc84-44b6-bf49-40f2d4cb11a8.png)

**Comment by gcangioli** [comment]
@costateixeira : is the publisher or the template that create the table ?

**Comment by costateixeira** [comment]
I understand that the table is made by the publisher java code

**Comment by Censacrof** [comment]
> @Censacrof : the code systems I looked at are value sets. In order to keep the different languages, unfortunately, you cannot simply change the declaration from CodeSystem to ValueSet and removing the definitions.
> 
> attached an example of how it can be done in sushi (just for one coded concept) [fsh-example.txt](https://github.com/hl7-eu/unicom-ig/files/10135279/fsh-example.txt)
> 
> Hope this helps :-)

Hi @gcangioli, i'm working on it. I have one question: if I create FHS files like in your example, when I compile with SUSHI, it counts them as Instances rather then ValueSet(s). Is this ok / desirable? I guess it doesn't make a difference in the end but I want to be sure.

**Comment by Censacrof** [comment]
@gcangioli why is copyright "CC-BY-SA-4.0"  in your example? Is this a mistake?

**Comment by Censacrof** [comment]
I created all ValueSets for UCUM, EDQM and SMS substances in #36 . 

Please @costateixeira @gcangioli @rlindstrm. Can you please review my work, and In case merge the pull request on the r4b so that we can close this issue?

**Comment by Censacrof** [comment]
@costateixeira @gcangioli @rlindstrm

I updated the EDQM valuesets so that they will only include localizations for only 7 (instead of 30+)  common languages. This resulted in way faster compilation time. Also, I fixed all the warnings saying that we were using languages that are not included in the "Common Languages" ValueSet.

I will now close this issue, as all essential ValueSets are in the IG and are working.
